File: src/connectors/base/categories/connector_categories_catalog.py
from .connector_category import ConnectorCategory
from common.py.utils import ItemsCatalog


@ItemsCatalog.define()
class ConnectorCategoriesCatalog(ItemsCatalog[ConnectorCategory]):
    """
    Global catalog of all connector categories.

    Categories are stored as fixed dataclasses describing each category.
    """

File: src/frontend/src/data/entities/connector/categories/ConnectorCategoriesCatalog.ts
import { ItemsCatalog } from "@common/utils/ItemsCatalog";

import { ConnectorCategory } from "./ConnectorCategory";

/**
 * Global catalog of all supported authentication schemes.
 */
@ItemsCatalog.define()
export class ConnectorCategoriesCatalog extends ItemsCatalog<ConnectorCategory> {}
